### PR TITLE
APEXMALHAR-1948: CassandraStore Should Allow You To Specify Protocol Version.

### PR DESCRIPTION
**Pull request by DT-Priyanka**

### CODE REVIEW

**Comment by ilooner** [comment]
@DT-Priyanka was this fix tested in an app on the cluster? I believe the hadoop jars pull in an older version of Guava, when the app is launched on cluster. Also please squash commits.

**Comment by DT-Priyanka** [comment]
@ilooner I haven't yet tested in on cluster. It will take time as our cluster doesn't have cassandra on it yet. This was review only, to see if code changes are fine. Once I test and squash changes would update review request accordingly.
Sorry forgot to add "review only"

File: contrib/src/main/java/com/datatorrent/contrib/cassandra/CassandraStore.java [hunk 1]
```diff
@@ -115,21 +130,35 @@ public Session getSession() {
...
     try {
-
-      cluster = Cluster.builder()
-          .addContactPoint(node).withCredentials(userName, password).build();
-    }
-    catch (DriverException ex) {
+      if (protocolVersion != null && protocolVersion.length() != 0) {
+        ProtocolVersion version = getCassandraProtocolVersion();
+        cluster = Cluster.builder().addContactPoint(node).withCredentials(userName, password).withProtocolVersion(version).build();
+      } else {
```
Comment (bhupeshchawda) [inline contrib/src/main/java/com/datatorrent/contrib/cassandra/CassandraStore.java:139]: which version does it use in the default case? Should we add a specific default? Like for example, the latest version?

File: contrib/pom.xml [hunk 2]
```diff
@@ -477,7 +477,14 @@
     <dependency>
       <groupId>com.datastax.cassandra</groupId>
       <artifactId>cassandra-driver-core</artifactId>
-      <version>2.0.2</version>
+      <version>2.1.8</version>
+      <optional>true</optional>
+    </dependency>
+    <dependency>
+      <groupId>com.google.guava</groupId>
```
Comment (bhupeshchawda) [inline contrib/pom.xml:493]: Why is this needed?

**Comment by bhupeshchawda** [comment]
Can you add unit tests if possible?

File: contrib/src/main/java/com/datatorrent/contrib/cassandra/CassandraStore.java (hunk 1 above)
Comment (DT-Priyanka) [inline contrib/src/main/java/com/datatorrent/contrib/cassandra/CassandraStore.java:139]: Default depends on cassandra driver version, with 2.1.x it is V2.

File: contrib/pom.xml (hunk 2 above)
Comment (DT-Priyanka) [inline contrib/pom.xml:493]: protocol version 3 was introduced in New in Java driver 2.1.2, that's why upgrading cassandra driver version.

**Comment by DT-Priyanka** [comment]
Added unit test as well as tested it on server.

**Comment by bhupeshchawda** [comment]
I am fine with the changes. I'll merge this in a couple of days if there are no other comments.
